### PR TITLE
refactor(staking): [LW-8510] refactor percentage slider

### PR DESCRIPTION
**Pull request by refi93**
# Checklist

- [x] JIRA - LW-8510
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

amend hotfix from #590 that fixed the infinite loop of the percentage slider, however it was at the cost of detaching the components from the `onPercentageChange` callback changes which makes it fragile/violates eslint rule on `useEffect()` hook exhaustive dependencies. This PR fixes that flaw by keeping the ref to the current callback which is updated anytime the callback is modified from above, without triggering any undesired `onPercentageChange()` callback invocations

On top of that, this PR gives more specific names to percentage-related props in the same component, suffixing them all consistently with the `percentage` suffix, improving overall readability of the code

## Testing

This PR should bring no noticeable change to the sliders' behavior, it's just a refactor. In order to test, open management drawer and try changing the slider values, check that after closing and reopening it still works as expected

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/2301/6379057980/index.html) for [0cf98d3e](https://github.com/input-output-hk/lace/pull/593/commits/0cf98d3e59f3776be83414d2fec604b577f39a1f)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 30     | 2      | 0       | 0     | 32    | ❌     |

### CODE REVIEW

File: packages/staking/src/features/drawer/preferences/PoolDetailsCard/PoolDetailsCard.tsx [hunk 1]
```diff
@@ -37,24 +37,28 @@ export const PoolDetailsCard = ({
...
+  const firstRender = useRef(true);
   const { t } = useTranslation();
-  const [localValue, setLocalValue] = useState(targetRatio);
+  const [percentage, setPercentage] = useState(targetPercentage);
+  const onPercentageChangeRef = useRef(onPercentageChange);
 
   useEffect(() => {
-    if (!effectInitialized.current) {
-      effectInitialized.current = true;
+    onPercentageChangeRef.current = onPercentageChange;
```
Comment (refi93) [inline packages/staking/src/features/drawer/preferences/PoolDetailsCard/PoolDetailsCard.tsx:52]: the main fix in this PR, took inspiration from the slider implementation in the lib

[state management hook used in the slider](https://github.com/radix-ui/primitives/blob/c31c97274ff357aea99afe6c01c1c8c58b6356e0/packages/react/slider/src/Slider.tsx#L102)
[the implementation of it](https://github.com/radix-ui/primitives/blob/c31c97274ff357aea99afe6c01c1c8c58b6356e0/packages/react/use-controllable-state/src/useControllableState.tsx#L12)
[the ref to the callback](https://github.com/radix-ui/primitives/blob/c31c97274ff357aea99afe6c01c1c8c58b6356e0/packages/react/use-callback-ref/src/useCallbackRef.tsx#L7)

File: packages/staking/src/features/drawer/preferences/PoolDetailsCard/PoolDetailsCard.tsx [hunk 2]
```diff
@@ -37,24 +37,28 @@ export const PoolDetailsCard = ({
...
+  }, [onPercentageChange]);
+
+  useEffect(() => {
+    if (firstRender.current) {
+      firstRender.current = false;
       return;
     }
-    onPercentageChange(localValue);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [localValue]);
```
Comment (refi93) [inline packages/staking/src/features/drawer/preferences/PoolDetailsCard/PoolDetailsCard.tsx:57]: actually, even the previous solution should work properly in the sense that if the value of the slider changed, the latest `onPercentageChange` passed from above would be used, but it's not very clean/intuitive that the `onPercentageChange` callback is left from the hook dependencies

**Review by szymonmaslowski — CHANGES_REQUESTED** [review]

File: packages/staking/src/features/drawer/preferences/PoolDetailsCard/PoolDetailsCard.tsx (hunk 1 above)
Comment (szymonmaslowski) [reply packages/staking/src/features/drawer/preferences/PoolDetailsCard/PoolDetailsCard.tsx:52]: I just realized we don't need `useEffect` at all. We removed a debounce so now we can just assign the `onPercentageChange` directly to the  Slider's `onValueChange`.

Comment (refi93) [reply packages/staking/src/features/drawer/preferences/PoolDetailsCard/PoolDetailsCard.tsx:53]: > I just realized we don't need useEffect at all. We removed a debounce so now we can just assign the onPercentageChange directly to the Slider's onValueChange.

indeed 🤦  good catch 👍

Comment (refi93) [reply packages/staking/src/features/drawer/preferences/PoolDetailsCard/PoolDetailsCard.tsx]: done in https://github.com/input-output-hk/lace/pull/593/commits/6a208d05973b38626059d3c2921b42999ea606ca

**Review by szymonmaslowski — APPROVED** [review]

**Review by przemyslaw-wlodek — APPROVED** [review]

**Review by oldGreg5 — APPROVED** [review]

**Pull request merged**